### PR TITLE
Add support for multiple user-defined specs

### PR DESCRIPTION
**Pull request by nickroess**
This PR changes `user_func_spec` from an option to a list to allow multiple user specs to be provided at once. Not ready for merge yet, I'm going to add a test

### CODE REVIEW

File: wp/plugin/wp.ml [hunk 1]
```diff
@@ -218,7 +218,7 @@ let func_name_map = Cmd.parameter Typ.(list ~sep:';' (pair ~sep:',' string strin
            "<reg1_orig>,<reg1_mod>;<reg2_orig>,<reg2_mod>"). By default, WP
            assumes subroutines have the same names between the two binaries.|}
 
-let user_func_spec = Cmd.parameter Typ.(some (t3 string string string)) "user-func-spec"
+let user_func_spec = Cmd.parameter Typ.(list ~sep:';' (t3 string string string)) "user-func-spec"
     ~doc:{|Creates the weakest precondition for a subroutine given the
```
Comment (fortunac) [inline wp/plugin/wp.ml:222]: Can you add how to write multiple user-defined specs and and an example?

File: wp/plugin/wp.ml [hunk 2]
```diff
@@ -218,7 +218,7 @@ let func_name_map = Cmd.parameter Typ.(list ~sep:';' (pair ~sep:',' string strin
            "<reg1_orig>,<reg1_mod>;<reg2_orig>,<reg2_mod>"). By default, WP
            assumes subroutines have the same names between the two binaries.|}
 
-let user_func_spec = Cmd.parameter Typ.(some (t3 string string string)) "user-func-spec"
+let user_func_spec = Cmd.parameter Typ.(list ~sep:';' (t3 string string string)) "user-func-spec"
```
Comment (fortunac) [inline wp/plugin/wp.ml:221]: Should it still be `user-func-spec` or should it be changed to `user-func-specs` (with the `s` at the end, or does it not matter)? Does anyone have thoughts?

File: wp/plugin/wp.ml (hunk 1 above)
Comment (fortunac) [reply wp/plugin/wp.ml:222]: I also realized that we don't have that flag documented on our site. Can you add it here? https://github.com/draperlaboratory/cbat_tools/blob/master/docs/reference.html

**Comment by fortunac** [comment]
Looks good so far! Can't wait to see the tests.

File: wp/plugin/wp.ml (hunk 2 above)
Comment (codyroux) [reply wp/plugin/wp.ml:221]: I have thoughts: it does matter (but not a lot!), and it should be `user_func_specs`.

Comment (codyroux) [reply wp/plugin/wp.ml:221]: or `user-func-specs` rather, though I'd probably rename the OCaml as well.

**Comment by nickroess** [comment]
@codyroux @fortunac Okay, just updated to
1) Rename `user-func-spec` to `user-func-specs` including all the examples that use the flag
2) Add test for multiple specs that needs 2 specs to get `UNSAT`
3) Update docs with example of multiple specs and added `user-func-specs` to `reference.html`

File: wp/resources/sample_binaries/user_func_spec/sub_spec_5/run_wp_1.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Should return SAT: we provide a spec only for g, which means f can still return 0x67 and we can trip the assert
+
+run () {
+  bap wp \
+    --func=main \
+    --precond="(assert (= (bvand RDI #xFFFFFFFF00000000) #x0000000000000000))" \
```
Comment (codyroux) [inline wp/resources/sample_binaries/user_func_spec/sub_spec_5/run_wp_1.sh:6]: Do you need these?

Comment (nickroess) [reply wp/resources/sample_binaries/user_func_spec/sub_spec_5/run_wp_1.sh:6]: Nope! I copied from other tests. Could be removed from this one and others

Comment (codyroux) [reply wp/resources/sample_binaries/user_func_spec/sub_spec_5/run_wp_1.sh:6]: Probably good not to have useless annotations in tests.

Comment (nickroess) [reply wp/resources/sample_binaries/user_func_spec/sub_spec_5/run_wp_1.sh]: K they are removed

**Pull request merged**